src/ray/gcs/gcs_client/service_based_accessor.h

```diff
@@ -43,7 +45,11 @@ class ServiceBasedJobInfoAccessor : public JobInfoAccessor {
       const SubscribeCallback<JobID, JobTableData> &subscribe,
       const StatusCallback &done) override;
 
+  Status AsyncReSubscribe() override;
+
  private:
+  SubscribeOperation subscribe_operation_;
```

Comment:
`// Save the subscribe operation in this function, so we can call it again when GCS restarts from a failure.`

Reply:
fixed